From 15387cba35abbe571199f7aa2b36503bc4f8f064 Mon Sep 17 00:00:00 2001
From: Phong Hoang <phong.hoang.wz@renesas.com>
Date: Fri, 22 Jul 2022 16:23:52 +0700
Subject: [PATCH 1/5] net: rswitch: Fix getting offset of GWTRCi register

Fix getting offset of GWTRCi register that is based on TX chain index.

Signed-off-by: Phong Hoang <phong.hoang.wz@renesas.com>
---
 drivers/net/rswitch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/rswitch.c b/drivers/net/rswitch.c
index c9dcee3e3b..47f2bdcc0b 100644
--- a/drivers/net/rswitch.c
+++ b/drivers/net/rswitch.c
@@ -1109,7 +1109,7 @@ static int rswitch_send(struct udevice *dev, void *packet, int len)
 	rswitch_flush_dcache((uintptr_t)desc, sizeof(*desc));
 
 	/* Start tranmission */
-	gwtrc_index = RSWITCH_TX_CHAIN_INDEX % 32;
+	gwtrc_index = RSWITCH_TX_CHAIN_INDEX / 32;
 	val = readl(gwca->addr + GWTRC(gwtrc_index));
 	writel(val | BIT(RSWITCH_TX_CHAIN_INDEX), gwca->addr + GWTRC(gwtrc_index));
 
-- 
2.25.1

